### ISSUE TITLE
[com.plumelabs.flow] Unable to log into an app

### ISSUE THREAD

**Issue by voxitme** [microg/android_packages_apps_GmsCore#842]
Hi there,

I'm unable to use a really specific app of my collection (Flow, package=[com.plumelabs.flow](https://play.google.com/store/apps/details?id=com.plumelabs.flow)).
The app launches correctly, but recquire immediatly to log in, and whereas there is absolutely no problem with this step on another GAPPS phone, it's been at least 3 times with 3 different roms I'm facing the same issue with microG: can't log in, having a loading circle looping infinitly.
Here is the corresponding logcat :

```
06-17 23:47:18.814 I/FirebasePerformance(19092): Session ID - 4aca52a6838f4ec981d9d07922d89453
06-17 23:47:18.815 D/Logger  (19092): (Application.kt:65) LoginActivity started
06-17 23:47:18.815 D/FirebasePerformance(19092): Creating a new Non Verbose Session: d60752291be14690aad0afb8ac72afb5
06-17 23:47:18.818 D/Logger  (19092): (Application.kt:61) LoginActivity resumed
06-17 23:47:18.821 W/ActivityManager(16239): Unable to start service Intent { act=com.google.android.gms.config.START pkg=com.google.android.gms } U=0: not found
06-17 23:47:18.821 W/ActivityManager(16239): Unbind failed: could not find connection for android.os.BinderProxy@7f37fd3
06-17 23:47:18.821 E/GmsClient(19092): unable to connect to service: com.google.android.gms.config.START on com.google.android.gms
06-17 23:47:18.824 D/GmsMeasureBrokerSvc(17687): onBind: Intent { act=com.google.android.gms.measurement.START pkg=com.google.android.gms }
06-17 23:47:18.826 D/GmsClearcutSvc(17687): onBind: Intent { act=com.google.android.gms.clearcut.service.START pkg=com.google.android.gms }
06-17 23:47:18.831 W/FirebaseRemoteConfig(19092): IPC failure: 17:API: Config.API is not available on this device.
06-17 23:47:18.835 D/SafeParcel(17687): Unknown field num 10 in com.google.android.gms.common.internal.GetServiceRequest, skipping.
06-17 23:47:18.835 D/SafeParcel(17687): Unknown field num 11 in com.google.android.gms.common.internal.GetServiceRequest, skipping.
06-17 23:47:18.835 D/SafeParcel(17687): Unknown field num 12 in com.google.android.gms.common.internal.GetServiceRequest, skipping.
06-17 23:47:18.835 D/GmsClearcutSvc(17687): bound by: GetServiceRequest{serviceId=CLEARCUT_LOGGER, gmsVersion=12451000, packageName='com.plumelabs.flow', extras=Bundle[{}]}
06-17 23:47:18.839 D/GmsClearcutLogSvcImpl(17687): log: LogEventParcelable[1, PlayLoggerContext[1, package=com.plumelabs.flow, packageVersionCode=78, logSource=-1, uploadAccount=null, loggingId=-1, logAndroidId=false, logSourceName=FIREPERF, isAnonymous=true, qosTier=0], LogEventBytes: CKLDoru2LTKjAQpoCigxOjEwMzE1NTYzODIzMzc6YW5kcm9pZDo4ZDliZGY3NDZkZjMxMTM1EgtkbVRLWU1iU3ZXMBotChJjb20ucGx1bWVsYWJzLmZsb3cSDzEuMC4wLjIyNTA1MzI1NhoGMS4yLjAzKAMSNwoDX2JzINi02vC+8eICKK/2ghVKIgogZDYwNzUyMjkxYmUxNDY5MGFhZDBhZmI4YWM3MmFmYjV4wHCIAaKj9CE=, AddPhenotypeExperimentTokens: true]
06-17 23:47:18.843 I/LatinIME(16521): Starting input. Cursor position = 46,46
06-17 23:47:18.847 D/SafeParcel(17687): Unknown field num 10 in com.google.android.gms.common.internal.GetServiceRequest, skipping.
06-17 23:47:18.847 D/SafeParcel(17687): Unknown field num 11 in com.google.android.gms.common.internal.GetServiceRequest, skipping.
06-17 23:47:18.847 D/SafeParcel(17687): Unknown field num 12 in com.google.android.gms.common.internal.GetServiceRequest, skipping.
06-17 23:47:18.847 D/GmsMeasureBrokerSvc(17687): bound by: GetServiceRequest{serviceId=MEASUREMENT, gmsVersion=12451000, packageName='com.plumelabs.flow', extras=Bundle[{}]}
06-17 23:47:18.848 D/GmsMeasureSvcImpl(17687): onTransact [unknown]: 10, android.os.Parcel@4f6de2e, 16
```

**Phone Model : OnePlus6**

![attachment1-2019-06-19](https://user-images.githubusercontent.com/36687040/59735356-8a333d00-9255-11e9-9dc9-44b6c7ffe9d9.jpeg)

**MicroG installation :**
![attachment2-2019-06-19](https://user-images.githubusercontent.com/36687040/59735358-8a333d00-9255-11e9-9f2b-f7db28c75398.jpeg)
Using the NanoDroid installer (NanoDroid-21.0.20190601)

Has anyone an idea on how to troubleshoot this case ? That might be related to #404 #529 #497 and #310 but not so much information there since :
- issue seems to have magically resolved with new release or
- issue was related to a -dirty microG build or
- user has got around not using the problematic app anymore

At this point, any help is welcome ;)

**Comment by Nanolx** [comment]
Not related to the actual issue, but: you're not on the latest unofficial beta build (which is 0.2.7.14755-beta-80). See https://www.nanolx.org/fdroid/repo

Additionally the next version of NanoDroid will by default install the official microG build instead of the unofficial custom build (see https://gitlab.com/Nanolx/NanoDroid/commit/8e115c0dbc62281e336cc5c06c25e05733b21b22) and provide the mapbox instead of vtm build (see https://gitlab.com/Nanolx/NanoDroid/commit/1f57ba0567b313e78762354d64386d25fac3667e).

(another step to get rid of the unofficial build)

**Comment by voxitme** [comment]
Still the same with version 0.2.7.17455-beta-80 (sorry for not having tested before, had a F-Droid cache issue, and thus didn't see the update earlier) unfortunately.
I'm gonna wait for the next NanoDroid release.

If ever you have the time for this, I'm having 2 questions by the way:

- Can we just flash a NanoDroid new version right over another, or should we always uninstall any previous one and clean install a new one? Asking this because of all the already GCM registered apps.
- If not, is there a way to reset the GCM registering process without having to backup data, reinstall and restore data?

Thanks by advance for any answer.

![2](https://user-images.githubusercontent.com/36687040/59811535-a93fd680-930a-11e9-935d-8e09d7fd7e18.jpeg)
![1](https://user-images.githubusercontent.com/36687040/59811538-aba23080-930a-11e9-9181-407dd934cd16.jpeg)

**Comment by Nanolx** [comment]
a) Why do you expect an uninstallation to be necessary? Just flash the update.
b) Yes, there is, if your device is rooted, run the following command in a shell: `nutl -r`  (reset GCM connection for all apps) or `nutl -r <appid>` , e.g. `nutl -r com.android.vending` (reset GCM connection for specific app)

**Comment by voxitme** [comment]
Ok about the updating process, thanks.
Oh yeah I'm sorry man I remember I saw it in the wiki but when I wanted to try it out previously (on another ROM), running `su` and then `nutl -r` , I had a "No command found" and couldn't figure out which package was necessary or what I did wrong.
It's now working as expected, don't know what is different from previous installation, but nevermind I got it now.
Thank you so much. Is this command still gonna work with official microG instead of your custom one ?

**Comment by voxitme** [comment]
I don't think it's gonna bring more information, but I eventually tried with the 0.2.7.17455 official microG GMS (using `nanodroid_gmscore=1` in the .nanodroid-setup config file) just in case, but sill the same :

> 06-21 02:09:35.227 W/BiChannelGoogleApi(31968): [FirebaseAuth: ] getGoogleApiForMethod() returned Gms: com.google.firebase.auth.api.internal.zzal@e3d1fb3

**Comment by Nanolx** [comment]
There's no (more) technical difference between both builds thus the error log is the same and yes `nanodroid-util` (or the `nutl` shortcut) works with both.

**Comment by voxitme** [comment]
Ok, thanks for those precisions.
Hoping it won't be too much pain, I really want to thank you @mar-v-in and @Nanolx for all the hard work done so far to bring everyone this so useful and great alternative.
Have a nice weekend :)

**Comment by starbrights** [comment]
I am reading here because of https://github.com/microg/android_packages_apps_GmsCore/issues/809
I am wondering howto find the problem in the many lines of locgat.

**Comment by voxitme** [comment]
> I am reading here because of #809
> I am wondering howto find the problem in the many lines of locgat.

If you strip out the logcat to the actual moment where the issue showed up (like few sec around it), it's suddenly easier to find the problem ;)
Also, it's easier if you have experience in debugging, and even more when you are the dev :D

**Comment by voxitme** [comment]
With new version [0.2.8.17785-beta-1 (8b4852b-mapbox)], I got this when trying to log in, with still the loading circle looping forever :

`07-04 15:17:10.658 W/BiChannelGoogleApi(21165): [FirebaseAuth: ] getGoogleApiForMethod() returned Gms: com.google.firebase.auth.api.internal.zzal@17578b1`

**Comment by x37-potter** [comment]
Hello, 
I am having the same issue on 0.2.8.17785-mapbox on the N26 app, can't go pass login.
Getting this error:
`[GmsClient: unable to connect to service: com.google.android.gms.config.START on com.google.android.gms]`

**Comment by mar-v-in** [comment]
@voxitme microG does not support Firebase Auth so far, which very likely is the issue here. I will let you know once first support is there.

@x37-potter The log you presented is most likely not the issue causing you the problem. Can you provide further logs?

**Comment by x37-potter** [comment]
Here you go, thank you so much for your help!
[log-number26.txt](https://github.com/microg/android_packages_apps_GmsCore/files/3369861/log-number26.txt)

**Comment by ale5000-git** [comment]
We need a dummy `com.google.android.gms.config.START`, I see it also in other issues.

**Comment by voxitme** [comment]
@x37-potter I personally never had an issue with the N26 app, Magisk not even being hidden to it...
Is it correctly registered in GCM ? Are other apps registered ?
If not (if not saying "Current state: connected since x" and not showing any apps list in the GCM menu), try this : go to the phone dialer and type *#*#2432546#*#* (or *#*#CHECKIN#*#*). Go back to microG and check if current state is now saying connected (don't remember if a reboot is necessary, maybe do it just in case). If yes, just uninstall reinstall N26 and all your other apps using GCM. They will appear in a list there.
Maybe that's useless for you but can help some others.

All of this is in the Nanolx's Nanodroid wiki.

**Comment by x37-potter** [comment]
Thanks, I didn't know about that wiki. Nope, I didn't manage to register it - the code didn't work (it said connected but no apps underneath).
Anyway, at the moment I had to uninstall microg because my phone becomes very slow and most apps don't start... I don't know what I did wrong.

**Comment by ildar** [comment]
> go to the phone dialer and type *#*#2432546#*#* (or *#*#CHECKIN#*#*).

misguiding. Right is: `**#**#2432546#**#**` (or `**#**#CHECKIN#**#**`)

**Comment by starbrights** [comment]
long time no updates, is it under development or dropped?

**Comment by voxitme** [comment]
Hey there,
I'm upping this post because I still have the issue after updating to v0.2.11.202414
Is there any information about this issue? I can provide a logcat if needed.

**Comment by saradasama** [comment]
> Hey there,
> I'm upping this post because I'm still having this issue after updating to v0.2.11.202414
> Is there any information about this issue? I can provide a logcat if needed.

I am also very curious about this specific issue as the app still does not let the user log in even with the latest version of microG. Should I provide a log about this behaviour?

**Comment by mar-v-in** [comment]
This should be fixed in c3bfb63f27c116d21ba7b2b87629e36573d4a59e (otherwise please reply).
